client/components/embed-container/index.jsx

```diff
@@ -112,6 +131,56 @@ function embedTumblr( domNode ) {
 	}, 30 );
 }
 
+function triggerJQueryLoadEvent() {
+	// force JetpackSlideshow to initialize, in case navigation hasn't caused ready event on document
+	window.jQuery( 'body' ).trigger( 'post-load' );
+}
+
+function createSlideshow() {
+	if ( window.JetpackSlideshow ) {
+		triggerJQueryLoadEvent();
+	}
+
+	loadAndRun( SLIDESHOW_URLS.JS, () => {
+		triggerJQueryLoadEvent();
+	} );
+}
+
+function embedSlideshow( domNode ) {
+	debug( 'processing slideshow for', domNode );
+
+	// set global variable required by JetpackSlideshow
+	window.jetpackSlideshowSettings = {
+		spinner: SLIDESHOW_URLS.SPINNER
+	};
+
+	// load CSS if it isn't alread there
+	if ( ! document.head.querySelector( `link[href="${ SLIDESHOW_URLS.CSS }"]` ) ) {
+		loadCSS( SLIDESHOW_URLS.CSS );
+	}
+
+	// Remove no JS warning so user doesn't have to look at it while several scripts load
+	const warningElements = domNode.parentNode.getElementsByClassName( 'jetpack-slideshow-noscript' );
+	forEach( warningElements, ( el ) => {
+		el.classList.add( 'hidden' );
+	} );
+
+	if ( window.jQuery && ! window.jQuery.prototype.cycle ) {
+		// Only jQuery exists
+		loadAndRun( SLIDESHOW_URLS.CYCLE_JS, ()=> {
+			createSlideshow();
+		} );
+	} else if ( window.jQuery && window.jQuery.prototype.cycle ) {
+		// jQuery and cylcejs exist
+		createSlideshow();
+	} else {
+		// Neither exist
+		loadjQueryDependentScript( SLIDESHOW_URLS.CYCLE_JS, () => {
```

Comment:
Similar to ```loadAndRun```, ```loadjQueryDependentScript``` already [checks to see if a request  to the same url is already in progress](https://github.com/Automattic/wp-calypso/blob/master/client/lib/load-script/index.js#L17-L35) by keeping track of currently loading scripts.